### PR TITLE
Auto cookie update

### PR DESCRIPTION
**Pull request by datawhores**
1.   optional auto-import of cookies
2.  support all browsers listed on browser_cookie3 Chrome, Firefox, LibreWolf, Opera, Opera GX, Edge, Chromium, Brave, Vivaldi, and Safari.
3. remove get_forum_cookies and change to get_browser_cookies  from other branch
4. add the fix  for the issue causing chrome and similar browsers to fail due to changes in sqlite. 
      browser_cookie3 has not been updated in a year

### CODE REVIEW

**Review by jbsparrow — APPROVED** [review]

**Comment by jbsparrow** [comment]
@datawhores Update poetry and re-generate lock file. You're on poetry version 1.4.1, most recent release is 1.8.4 😿

**Review by jbsparrow — APPROVED** [review]

File: poetry.lock [hunk 1]
```diff
@@ -1,9 +1,10 @@
-# This file is automatically @generated by Poetry 1.8.4 and should not be changed by hand.
+# This file is automatically @generated by Poetry 1.4.1 and should not be changed by hand.
```
Comment (jbsparrow) [inline poetry.lock:1]:
```suggestion
# This file is automatically @generated by Poetry 1.4.1 and should not be changed by hand.
```

Upgrade to poetry 1.8.4.

**Review by jbsparrow — CHANGES_REQUESTED** [review]

File: poetry.lock (hunk 1 above)
Comment (datawhores) [reply poetry.lock]: done

**Review by jbsparrow — DISMISSED** [review]

**Comment by jbsparrow** [comment]
Also for ruff, you have to manually run it. If you have ruff installed you just run `ruff format` and `ruff check --fix` in the root directory of the project to automatically format everything according to the rules set out in pyproject.toml.

**Review by NTFSvolume — CHANGES_REQUESTED** [review]
Install the pre-commit hooks or run ruff manually to fix the lint errors.

**Comment by jbsparrow** [comment]
Ohhh you have to install them 😭 that makes sense 😭

**Comment by jbsparrow** [comment]
> Install the pre-commit hooks or run ruff manually to fix the lint errors.

Ohh 😭 That makes sense I didn't know that you had to install them 💀

File: cyberdrop_dl/utils/args/browser_cookie_extraction.py [hunk 2]
```diff
@@ -39,22 +41,20 @@ def wrapper(*args, **kwargs) -> CookieJar:
...
 
+def get_cookies_from_browser(manager: Manager, browser: str=None) -> None:
+    """Get the cookies for the supported sites"""
+    manager.path_manager.cookies_dir.mkdir(exist_ok=True)
+    browser=browser or manager.config_manager.settings_data["Browser_Cookies"]["browser"]
+    browser=browser.lower()  if browser else None
+    for domain in SupportedDomains.supported_forums:
+        cookies = get_cookie(browser, domain)
+        cookie_jar = MozillaCookieJar()
+        cookie_file_path = manager.path_manager.cookies_dir / f"{domain}.txt"
```
[the remark this answers is not in the document]
Comment (jbsparrow) [reply cyberdrop_dl/utils/args/browser_cookie_extraction.py:52]: Can't we instead, on the first run after this is implemented, iterate through all supported browsers, and pull cookies a popular website (or websites), then compare to see which browser has the most recent last accessed date?

That seems overcomplicated now that I really think about it. What about just having no default for it, if the user enables auto-import, we prompt them to choose their browser on startup.

**Comment by NTFSvolume** [comment]
> Ohhh you have to install them 😭 that makes sense 😭

Install with : `poetry run pre-commit install`

File: cyberdrop_dl/utils/args/browser_cookie_extraction.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (NTFSvolume) [reply cyberdrop_dl/utils/args/browser_cookie_extraction.py:52]: That option was also added. The user can change it but it defaults to chrome

**Review by NTFSvolume — APPROVED** [review]
Will merge as is. Ruff formating can be fix later on

**Pull request merged**